aws/resource_aws_alb.go

```diff
@@ -401,6 +402,62 @@ func resourceAwsAlbDelete(d *schema.ResourceData, meta interface{}) error {
 		return fmt.Errorf("Error deleting ALB: %s", err)
 	}
 
+	err := cleanupALBNetworkInterfaces(meta.(*AWSClient).ec2conn, d.Id())
+	if err != nil {
+		log.Printf("[WARN] Failed to cleanup ENIs for ELB %q: %#v", d.Id(), err)
+	}
+
+	return nil
+}
+
+// ALB automatically creates ENI(s) on creation
+// but the cleanup is asynchronous and may take time
+// which then blocks IGW, SG or VPC on deletion
+// So we make the cleanup "synchronous" here
+func cleanupALBNetworkInterfaces(conn *ec2.EC2, albArn string) error {
+	re := regexp.MustCompile("([^/]+/[^/]+/[^/]+)$")
+	matches := re.FindStringSubmatch(albArn)
+	if len(matches) != 2 {
+		return fmt.Errorf("Unexpected ARN format: %q", albArn)
+	}
+
+	// e.g. app/example-alb/b26e625cdde161e6
+	name := matches[1]
+
+	out, err := conn.DescribeNetworkInterfaces(&ec2.DescribeNetworkInterfacesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("attachment.instance-owner-id"),
+				Values: []*string{aws.String("amazon-elb")},
+			},
+			{
+				Name:   aws.String("description"),
+				Values: []*string{aws.String("ELB " + name)},
```

Comment:
This is actually intention - the underlying API uses a different naming convention, "ELB v2" they call it, so ELB is correct here in this context.